tests/unit/test_builder.py

```diff
@@ -797,6 +808,7 @@ def test__fetch_config_image_name(config: builder.FetchConfig, expected_name: st
                     microk8s_channels=("",),
                     prefix="",
                     runner_version="",
+                    script_url=None,
```

Comment:
Do you think we should have a test where the script url is not `None`?

Reply:
Would it be alright to follow up on the datastructures? I am going to refactor the state and the builder data DTOs. The `script_url` parameter does nothing here and isn't worth testing different values. It indicates that the function and the DTOs require refactoring.

Reply:
The refactor will:
1. Merge all the charm state values
2. Differentiate charm state values into 3 data wrappers:
  - App configuration parameters(build interval, channel, number of parallel build, app resource prefix)
  - Image configuration parameters(architecture, juju channel, microk8s channel, OS bases, ..etc)
  - Cloud configuration parameters(build cloud, upload cloud)
  - External service configuration parameters(dockerhub, proxy, ..etc)
 3. Create a DTO layer between the state and the `builder.py` module (to decouple from charm state)
 4. Charm translates state models into builder data structures.